src/Microsoft.DotNet.Cli.Utils/CommandResolution/PublishPathCommandSpecFactory.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class PublishPathCommandSpecFactory : IPublishedPathCommandSpecFactory
+    {
+        public CommandSpec CreateCommandSpecFromPublishFolder(
+            string commandPath,
+            IEnumerable<string> commandArguments,
+            CommandResolutionStrategy commandResolutionStrategy,
+            string depsFilePath,
+            string runtimeConfigPath)
+        {
+            return CreateCommandSpecWrappingWithMuxerIfDll(
+                commandPath,
+                commandArguments,
+                depsFilePath,
+                commandResolutionStrategy,
+                runtimeConfigPath);
+        }
+
+        private CommandSpec CreateCommandSpecWrappingWithMuxerIfDll(
+            string commandPath,
+            IEnumerable<string> commandArguments,
+            string depsFilePath,
+            CommandResolutionStrategy commandResolutionStrategy,
+            string runtimeConfigPath)
+        {
+            var commandExtension = Path.GetExtension(commandPath);
+
+            if (commandExtension == FileNameSuffixes.DotNet.DynamicLib)
+            {
+                return CreatePackageCommandSpecUsingMuxer(
+                    commandPath,
+                    commandArguments,
+                    depsFilePath,
+                    commandResolutionStrategy,
+                    runtimeConfigPath);
+            }
+
+            return CreateCommandSpec(commandPath, commandArguments, commandResolutionStrategy);
```

Comment:
Is there a case where we'll be running arbitrary executables from the publish directory? desktop?